site/app/templates/grading/electronic/Details.twig

```diff
@@ -697,5 +708,82 @@
         }
         location.reload();
     }
+
+    //function to count how many submissions are being regraded so that a confirm message can be displayed if necessary
+    function ajaxCountRegrade(choice,e,confirmMessage) {
+        const submit_url = buildCourseUrl(['gradeable', "{{ gradeable.getId }}", 'regrade', 'count']);
+        let formData = new FormData();
+        formData.append('csrf_token', csrfToken);
+        formData.append('gradeable_id', "{{ gradeable.getId }}");
+        //if regrading all versions
+        if (choice) {
+            formData.append('regrade_all', 'true');
+        }
+        else {
+            formData.append('regrade_all', 'false');
+        }
+        $.ajax({
+            url: submit_url,
+            data: formData,
+            processData: false,
+            type: 'POST',
+            headers : {
+                Accept: "application/json"
+            },
+            contentType: false,
+            success: function(data) {
+                data = JSON.parse(data);
+                console.log(data);
+                if (data['status'] === 'success') {
+                    confirmMessage(e, data['data'], choice);
+                }
+                else {
+                    if (data['message'] === "You do not have access to that page.") {
+                        window.location.href = '';
+                    }
+                    else {
+                        alert("ERROR! Please contact administrator with following error:\n\n" + data['message']);
+                    }
+                }
+            }
+        });
+    }
+    function confirmMessage(e,count, choice) {
+        if(count > 50) {
+            e.preventDefault();
```

Comment:
Why prevent the default only for `count > 50`? What is the default that we're preventing here?

Reply:
Barb wanted a confirm message to pop up if the instructor attempted to regrade over 50 submissions, to make sure they don't accidentally send a bunch of files to the to be graded queue